### DISCUSSION TITLE
Multiple Responses from Destination (TCP Sender)

### DISCUSSION THREAD

**Discussion by ckar69** [nextgenhealthcare/connect#4946] [General]
Hi,
I have a channel with a TCP Sender Destination. Destination app is sending back one message (ACK.1) as a Communication ACK and after a while (when the message is processed) tries to send an Application ACK (ACK.2). 
Destination response though is the first message. 
When the Destination is triggered again with another message, then Destination response is the ACK.2 of the previous message.
The TCP Sender Destination has "Keep Connection Open".

I don't know how to get the 2 messages for every single message I sent to the Destination.

It seems that a similar subject was mentioned here:
https://forums.mirthproject.io/forum/mirth-connect/support/16462-accepting-multiple-response-in-tcp-mllp#post93248

I would appreciate any possible help.

**Comment by jonbartels** [comment]
Do you have any way to inspect the message structure of the response? Maybe a packet capture in addition to MC. I'm interested in seeing the control characters, if any, for each message.

I am wondering if there are two distinct messages or if the response looks something like an HL7 batch message.

Can you say what system you are talking to?
What kind of message are you sending? Is it one of the HL7 Qxx types?

**Comment by ckar69** [comment]
Hi, the response is two HL7 messages written in the same stream.
<VT>Message Data<FS><CR>
TCP Sender Destination's response gets only the first (Communication ACK).
It has to wait to get the next message that is the Application ACK (after the message is processed).
I guess Mirth is waiting only one message (gets <FS>CR> characters) and never reads from stream again, untill it sends the next message to the destination.

When it does, it reads the message from the stream, which is actually the Application ACK, belonging (business wise) to the first message sent.

**Comment by odoodo** [comment]
Just add a 2nd destination to your channel. Resend the request or any dummy request via this destination and use the response - which should now be the real payload - for your further processing.
